### PR TITLE
✨ Approve amount of selected token in the swap panel + fetch balances

### PR DESCRIPTION
**Pull request by YohanTz**
## Related Issues

#66, #67

## Before
Screenshot 2023-11-10 at 16 05 40

## After
https://github.com/Zohal-Starknet/zohal-interface/assets/37301269/d6645c1c-c915-4279-993f-0136ee990a27

## Checklist
- [x] ✅ All existing tests pass.
- [x] 📃 Documentation has been updated (if necessary).
- [x] 🚀 This PR has been tested locally and is ready for review.

### CODE REVIEW

File: src/app/_helpers/tokens.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export const TOKENS: Record<
+  string,
+  { address: `0x${string}`; decimals: number; name: string }
```
Comment (akassaei) [inline src/app/_helpers/tokens.ts:3]: NIT : Move that into its proper type maybe ?

File: src/app/trade/_components/swap-action-button.tsx [hunk 2]
```diff
@@ -43,7 +53,11 @@ export default function SwapActionButton(props: SwapActionButtonProps) {
     );
   }
 
-  return <Button className={commonSwapActionButtonClassName}>Swap</Button>;
+  return (
+    <Button className={commonSwapActionButtonClassName} onClick={() => swap()}>
```
Comment (akassaei) [inline src/app/trade/_components/swap-action-button.tsx:57]: NIT : I think we can do that directly
```suggestion
    <Button className={commonSwapActionButtonClassName} onClick={swap}>
```

File: src/app/trade/_components/swap-input.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { type TokenSymbol } from "@zohal/app/_helpers/tokens";
+import Input from "@zohal/app/_ui/input";
+
+type SwapInputProps = {
+  formattedTokenBalance?: string;
```
Comment (akassaei) [inline src/app/trade/_components/swap-input.tsx:5]: NIT : JSDoc

**Review by akassaei — DISMISSED** [review]

File: src/app/trade/_components/swap-action-button.tsx (hunk 2 above)
Comment (YohanTz) [reply src/app/trade/_components/swap-action-button.tsx:57]: I Can't unfortunately, Button tries to pass an event to the function

File: src/app/_helpers/tokens.ts (hunk 1 above)
Comment (akassaei) [reply src/app/_helpers/tokens.ts:3]: I ment the one with address, decimals and name

**Review by akassaei — APPROVED** [review]

**Pull request merged**